pkg/driver/controller.go

```diff
@@ -241,11 +244,17 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			snapshotArn = &snapshot.ResourceARN
 		}
 
+		// When dynamically provisioning child volumes, we ignore the user-provided request for storage.
+		// The storage on the volume will be configured via the storageCapacityReservation and storageCapacityQuota
+		// parameters in the storage class. To prevent users from attempting to provision storage via this value,
+		// we force them to provide a default storage request on the PVC, which is 1GiB.
+		if *storageCapacity != util.DefaultVolumeStorageRequestGiB {
```

Comment:
The PVC only accepts a positive request for storage. I think forcing the user to enter 1GiB is the best option we have, but let me know what you think. Sam is pretty adamant that we enforce a specific value so the user is not misled.